src/store/index.ts

```diff
@@ -0,0 +1,750 @@
+      ipcRenderer.once("sRoomUpdated", () => {
+        commit("setLoading", false);
+        return true;
+      });
+
+      ipcRenderer.once("sRoomUpdateFailed", (event, data) => {
+        dispatch("errDialog", {
+          title: "Failed to update room",
+          body: data.reason,
+          canClose: true,
+        });
+        commit("setLoading", false);
+        return false;
+      });
+    },
+    leaveRoom: async () => {
+      commit("setLoading", true);
+      const key = getters.isHost.value ? "cRoomDelete" : "cLeaveRoom";
+      const payload = {
+        id: getters.roomId.value,
+        userId: getters.userId.value,
+      };
+      ipcRenderer.send("CLIENT_MESSAGE", { key, payload });
+      const evt = getters.isHost.value ? "sRoomDeleted" : "sUserLeftRoom";
+      ipcRenderer.once(evt, (event, data) => {
+        if (evt == "sUserLeftRoom") {
+          if (data.userId == getters.userId.value) {
+            commit("setLoading", false);
+            return true;
+          }
+        } else {
+          commit("setLoading", false);
+          return true;
+        }
+      });
+    },
+    kickUser: async (payload) => {
+      commit("setLoading", true);
+      ipcRenderer.send("CLIENT_MESSAGE", { key: "cKickUser", payload });
+      ipcRenderer.on("sUserKicked", () => {
+        commit("setLoading", false);
+        return true;
+      });
+    },
+    banUser: async (payload) => {
+      commit("setLoading", true);
+      ipcRenderer.send("CLIENT_MESSAGE", { key: "cBanUser", payload });
+      ipcRenderer.on("sUserBanned", () => {
+        commit("setLoading", false);
+        return true;
+      });
+    },
+    requestRooms: async (payload) => {
+      ipcRenderer.send("CLIENT_MESSAGE", {
+        key: "cRequestRoomList",
+        payload: { page: payload && payload > 0 ? payload : 0 },
+      });
+    },
+    sendChat: (payload) => {
+      ipcRenderer.send("CLIENT_CHAT", {
+        key: "cChat",
+        payload,
+      });
+      commit("pushChat", {
+        id: payload.id,
+        userId: state.user.id,
+        name: state.user.name.trim(),
+        message: payload.message.trim(),
+      });
+    },
+    sendFlags: () => {
+      const flags = getters.flags.value
+        .map((val) => {
+          let flag = { flag: val.id };
+          if (typeof val.value == "number") {
+            flag.uIntVal = val.value;
+          } //temp fix
+          if (val.type == "boolean" && !val.value) {
+            flag = undefined;
+          }
+          return flag;
+        })
+        .filter((v) => typeof v !== "undefined");
+      ipcRenderer.send("CLIENT_MESSAGE", {
+        key: "cRoomFlagsUpdate",
+        payload: { flags },
+      });
+    },
+    startRun: (payload) => {
+      ipcRenderer.send("CLIENT_MESSAGE", {
+        key: "cStartRun",
+        payload,
+      });
+    },
+  };
+
+  function commit(type: keyof typeof mutations, payload) {
+    return mutations[type](payload);
+  }
```

Comment:
This `commit` function and the one below only exists so that I didn't have to refactor quite as much. Eventually it can be removed.